src/main/java/lotto/domain/Lotto.java

```diff
@@ -1,25 +1,41 @@
 package lotto.domain;
 
+import java.util.List;
+import java.util.stream.Stream;
+
 public class Lotto {
 	private final LottoNumbers lottoNumbers;
 
-	public Lotto(LottoNumbers lottoNumbers) {
-		this.lottoNumbers = lottoNumbers;
+	public Lotto(final int... lottoNumbers) {
+		this(new LottoNumbers(lottoNumbers));
 	}
 
-	public Lotto(int... lottoNumbers) {
+	public Lotto(final String lottoNumbers) {
 		this(new LottoNumbers(lottoNumbers));
 	}
 
-	public int countMatchNumber(WinningNumbers winningNumbers) {
-		int count = 0;
-		for (Integer winningNumber : winningNumbers.value()) {
-			count += lottoNumbers.isMatch(winningNumber);
-		}
-		return count;
+	public Lotto(final LottoNumbers lottoNumbers) {
+		this.lottoNumbers = lottoNumbers;
+	}
+
+	public long countMatchedNumber(WinningLotto winningLotto) {
+		return this.lottoNumbers.stream()
+			.filter((lottoNumber) -> winningLotto.stream()
+				.anyMatch(winningNumber -> winningNumber.equals(lottoNumber)))
+			.count();
 	}
```

Comment:
현재 WinningLotto 클래스는 winningNumbers와 bonusBallNumber를 그저 담아두는 클래스로 사용되고 있는데요.
WinningLotto의 적절한 책임은 무엇일까요? 
아래 예시와 같이 WinningLotto를 통해 matchCount를 구할 수도 있고, 
근본적인 책임에 대해 생각해보면 우승 로또 (WinnigLotto)에 로또 1줄 (Lotto)를 매치하여 Rank를 반환하도록 해볼 수도 있을 것 같아요 :)

```suggestion
	public long countMatchedNumber(WinningLotto winningLotto) {
		return winningLotto.matchCount(lottoNumbers);
	}
```